lib/core/src/lib/notifications/components/notification-history.component.ts

```diff
@@ -81,20 +83,24 @@ export class NotificationHistoryComponent implements OnDestroy, OnInit, AfterVie
     }
 
     addNewNotification(notification: NotificationModel) {
-        this.notifications.unshift(notification);
+        this.badgeHidden = !this.isBadgeVisible();
+        this.unreadNotifications.unshift(notification);
 
-        if (this.notifications.length > NotificationHistoryComponent.MAX_NOTIFICATION_STACK_LENGTH) {
-            this.notifications.shift();
+        if (this.unreadNotifications.length > NotificationHistoryComponent.MAX_NOTIFICATION_STACK_LENGTH) {
+            this.unreadNotifications.shift();
         }
 
         this.saveNotifications();
         this.createPagination();
     }
 
     saveNotifications() {
-        this.storageService.setItem(NotificationHistoryComponent.NOTIFICATION_STORAGE, JSON.stringify(this.notifications.filter((notification) =>
-            notification.type !== NOTIFICATION_TYPE.RECURSIVE
-        )));
+        this.unreadNotifications.forEach((notification: NotificationModel) => {
+            this.notifications.push(notification);
+        });
+
+        this.storageService.setItem(NotificationHistoryComponent.NOTIFICATION_STORAGE, JSON.stringify(this.notifications));
```

Comment:
I see there are some usages of 'notification-history' in your PR for applications. Could you please move 'notification-history' to different file than component file (it may be completely new file or some existing like notification.model.ts) and then reused this string in both ADF and applications PRs?

Reply:
moved 'notification-history' to notification.model.ts

Reply:
Hi @AleksanderSklorz , After moving 'notification-history' to notification.model.ts, I am facing one issue in ADW. 
I have tried to run ADW with ADF build locally, and all the changes made in _'notification.model.ts'_ are visible in _'notification.model.d.ts'_ and functionality is working fine.

But while running the unit tests, I get this error` /libs/content-ee/content-services-extension/src/lib/components/notifications/content-extension-notification.service.spec.ts:37:31-51 - Error: export 'NOTIFICATION_STORAGE' (imported as 'NOTIFICATION_STORAGE') was not found in '@alfresco/adf-core'.
`
1. I have double-checked that _notification.model.ts_ is already being exported in _public-api.ts_ and 
2. another strange thing I observed is that instead of adding/exporting a new variable if I add a key-value pair to an existing enum in the _notification.mode.ts_ file then the tests work fine.

Could I be missing something here? 

Also, previously we were using 'NotificationHistoryComponent.NOTIFICATION_STORAGE', which worked fine because NotificationHistoryComponent is already an exported component (in develop). Should we go back to the previous approach?

Reply:
discussed on slack so closing this comment